src/mds/Server.cc

```diff
@@ -1212,9 +1223,20 @@ void Server::find_idle_sessions()
 	break;
       }
       to_evict.push_back(session);
+      if (any_osd_laggy) {
+	laggy_sessions.insert(session);
+      }
     }
   }
 
+  // don't kick client if osds are laggy
+  if(get_laggy_sessions()) {
+    dout(10) << __func__ << " detected laggy sessions due to laggy OSDs,"
+             << " skipping laggy clients eviction untill they return to normal"
+             << dendl;
```

Comment:
```suggestion
    dout(0) << "Detected " << laggy_sessions.size()
               << " unresponsive sessions, possibly due to laggy OSDs."
             " Eviction is skipped until the OSDs return to normal."
             << dendl;
```

Reply:
i'd also s/laggy_sessions/unresponsive_sessions

Reply:
as per https://github.com/ceph/ceph/pull/49971#discussion_r1145224688, it's more laggy sessions then, i think laggy clients and unresponsive clients would be better